src/controllers/MultiCacheController.ts

```diff
@@ -65,6 +65,74 @@ export default class MultiCacheController {
         .reflect();
     }
 
+    this.execAll(actions, reply);
+  }
+
+  public getFlushURLsConfig(cacheControllers: CacheController[]): Hapi.IRouteAdditionalConfigurationOptions {
+    return {
+      handler: this.flushURLs.bind(this),
+      tags: ['api', 'cache'],
+      description: 'Flush multiple caches',
+      validate: {
+        payload: this.getPayloadRequestSchema(cacheControllers, 'flushURLsConfig'),
+        failAction: (request: Hapi.Request, reply: Hapi.IReply, source: string, error: Boom.BoomError) => {
+          reply(ResponseBuilder.cleanupError(error));
+        },
+      },
+      plugins: {
+        'hapi-swagger': {
+          responses: {
+            200: {
+              description: 'All caches have been flushed',
+              schema: this.getSuccessResponseSchema(cacheControllers),
+            },
+            400: {
+              description: 'Bad Request',
+              schema: this.getBadRequestResponseSchema(cacheControllers),
+            },
+            502: {
+              description: 'Bad Gateway',
+              schema: this.getBadGatewayResponseSchema(cacheControllers),
+            },
+          },
+        },
+      },
+    };
+  }
+
+  public flushURLs(request: Hapi.Request, reply: Hapi.IReply) {
+    const actions: IMap<any> = {};
+    // tslint:disable:no-string-literal
+    if (request.payload['keycdn']) {
+      actions['keycdn'] = keycdn.flushURLs(request.payload['keycdn']['zoneID'],
+                                           request.payload['keycdn']['urls'],
+                                           request.payload['keycdn']['authorizationToken'])
+        .reflect();
+    }
+
```

Comment:
is fasterize pointing to fasterize cache or to future cloudfront CDN?

Reply:
This is our own API. We don't use it as this is our API which calls Darkstar but it was more an exercise when there was only KeyCDN to flush.